src/docfx/lib/schema/JsonSchemaValidation.cs

```diff
@@ -58,6 +58,24 @@ private static void ValidateScalar(JsonSchema schema, List<Error> errors, JValue
             {
                 errors.Add(Errors.UndefinedValue(JsonUtility.GetSourceInfo(scalar), scalar, schema.Enum));
             }
+
+            if ((schema.MaxLength.HasValue || schema.MinLength.HasValue) && CanTreatAsString())
```

Comment:
Is checking the value good enough to detect whether it is a string?

```suggestion
            if ((schema.MaxLength.HasValue || schema.MinLength.HasValue) && scalar.Value is string str)
```